### DISCUSSION TITLE
Is there a limit per day for fetching followers?

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#171] [Q&A] [answered]
Hello,

Is there a daily limit for fetching followers? I am using default instagram variable for safety with`Requester.default = .instagram` . However, while I was developing an account got banned. I guess I request 100 times (We can say called followers endpoint with .max page 10 times) in 1 hour.

**Comment by sbertix** [comment, marked as answer]
Hey @emreond, 

Followers/following accounts are heavily rate-limited, especially for accounts that are already marked as spam (for instance, if you use them mostly with **Swiftagram**, **SwiftyInsta** or any other non-Instagram client, this is very likely to happen). 
There's not an exact figure I can give you, unfortunately, as Instagram, allegedly, stopped relying on hard caps in favor of AI likelihood of a single event being "spam". 

However, I can say with confidence that 100 times/hour is definitely way too much. You need to understand the Private API are used for anything requiring user interaction and limits tend to follow that assumption. 

There's nothing I can do unfortunately to make this better: it's an inherent problem with **any** Instagram Private API client. For paginating followers/following I suggest relying on actual scraping (which is faster and is not rate limited) and the `UIKit`+`WebKit` implementation, with authentication cookies, is pretty straightforward.

**Reply by emreond** [reply]
Hmm thanks for your reply. I guess I will limit it with 3 times per day (fetching all followers).

**Reply by VitaliiDeveloper** [reply]
As I found, it doesn't matter how much time you request to fetch all followers. Does matter only count of paging before was opened. Tried to delay 1 sec per page but anyway, for a big account, it's a heavy operation. So you need to set a delay depending to the account followers count.

**Comment by Phoenix-Genius** [comment]
In my opinion, there is no limit on the number of followers, but I think you can only follow up to 2K users per day.
I followed 800+ in 2 hours and had no problems.
I use the free account, but I don't know how GitHub Pro account.
If you have any questions about other issues, feel free to ask.

**Reply by igornazarov1991** [reply]
Hey @sbertix, 

Thanks for your detail response and explanation of the situation. When you say 

> For paginating followers/following I suggest relying on actual scraping (which is faster and is not rate limited) and the UIKit+WebKit implementation, with authentication cookies, is pretty straightforward

do you mean loading the relevant Instagram web pages in a web view and then running JavaScript within that web view to get out the data you want? 

I'm still trying to find a way to login to Instagram via web view and then calling all needed endpoints natively. Theoretically it is possible to replicate what web version of Instagram is doing. What do you think?

The problem I'm facing right now is that after performing a set of request for followers, Instagram responds with the following error:

```
{
"message": "Please wait a few minutes before you try again.",
"require_login": true,
"status": "fail"
} 
```

I'm confused why web version doesn't have this issue. I've beed experimenting a lot with it and web is able to load followers by sending requests with time interval about two seconds. What is interesting is that after login web version sends 

`x-ig-www-claim`

with each subsequent request. This field is sent in the response for the login process. I assume this is a very important data. It is not so easy to intercept it from WKWebView (at least I didn't find a way at the moment). Do you know if it is possible to generate it locally relying on some other data?

Thanks in advance for any response!
